### ISSUE TITLE
Refresh dashboard for every change of image/model/XAI method.

### ISSUE THREAD

**Issue by cpranav93** [dianna-ai/dianna#378]
The dashboard currently only refreshes the results when a new XAI method is employed. However, it would be great to have the results refresh every time the image or the ONNX model is changed as well.

**Comment by laurasootes** [comment]
I am thinking the best solution might be to create some sort of "update" button to start the calculation. In #352 I added buttons for the various parameters of the different models. It is annoying that the dashboard refreshes as soon as you change one of them, instead of when you are finished setting all of them.

**Comment by geek-yang** [comment]
> I am thinking the best solution might be to create some sort of "update" button to start the calculation. In #352 I added buttons for the various parameters of the different models. It is annoying that the dashboard refreshes as soon as you change one of them, instead of when you are finished setting all of them.

Just to leave a note here. During the standup we decided to add buttons for update to avoid rapid refreshing if users tweak the parameters. But we agreed that we still need this refresh functionality for images/models/methods update. So now, we will need to review and merge the PR #352 first, and then combine the changes in the branch which addresses this issue (FYI @cpranav93).

**Comment by geek-yang** [comment]
Pranav and I tried to implement the refreshing functionality. Although it works in some cases, this makes the dashboard very unstable and could sometimes cause some "disorder" behavior (e.g. LIME triggered automatically without selection).

After some struggling, we decide to proceed with "update" button as a solution to this. It will also solve the problem for rapid refreshing in PR #387.

**Comment by cpranav93** [comment]
@geek-yang has implemented the update button for the dashboard. The idea behind is that the user has control on when they wish to refresh the results page and to perform the calculations. I have added some minor visual changes to it and also fixed an issue where the dianna calculations were performed when the method was being selected and a second time when the update button is pressed. Now nothing happens unless the update button is clicked. I think we can move this to review. @geek-yang, your thoughts?

**Comment by geek-yang** [comment]
Thanks for the changes @cpranav93. I just noticed that there is one failing test for the linter in our PR #396. After it is fixed, I think we can move it to the review column. I can take a look.

**Comment by cpranav93** [comment]
Hi @geek-yang, two issue still to be addressed (that I have found):
1) The error "Missing model, image or XAI method." is raised when the dashboard is new and one of these is missing. However, after running through the selection of everything once, then if you remove the method (for e.g.), then it does not raise the error. This should be fixed.
2) Currently linter raised some issues with variables in the compute_value_i path which we had commented out as it lead to dual computations - however, these actually use the global cache to check for the availability of results and thus should not be commented out. When you do add them, however, this leads to dual computations again. We have to fix this as well.

I leave it to you on whether you want to raise separate issues for these or try to tackle them here in this one.
